@xen-orchestra/proxy/src/app/mixins/api.mjs

```diff
@@ -118,7 +119,14 @@ export default class Api {
       .use(router.routes())
       .use(router.allowedMethods())
 
-    httpServer.on('request', koa.callback())
+    const callback = koa.callback()
+    httpServer.on('request', (req, res)=>{
+      // only answers to query to the root url of this mixin
+      // do it before giving the request to Koa to ensure it's not modified
+      if(req.url.startsWith(PREFIX)){
+        callback(req, res)
+      }
+    } )
```

Comment:
Please answer this.

Reply:
What other  mixin are you talking about ? 

Most of the logic of the proxy is in the onReq method, Koa would not have simplified notably the code , and would have added another layer.

Reply:
I feel like current implementation makes it difficult for mixins to handle the requests they want and also to have a proper `404` error in case a path is not handled.

But let's keep this like this for now.